lib/bunq/resource.rb

```diff
@@ -21,11 +35,13 @@ def get(params = {}, &block)
       raise Bunq::Timeout
     end
 
-    def post(payload, skip_verify: false, encrypt: false, &block)
-      body = JSON.generate(payload)
+    def post(payload, skip_verify: false, encrypt: false, custom_headers: {}, &block)
+      custom_headers = JSON.parse(custom_headers.to_json)
+      body = post_body(payload, custom_headers)
       body, headers = client.encryptor.encrypt(body) if encrypt
+      headers = headers.to_h.merge(custom_headers)
```

Comment:
Can `headers` be blank? So maybe add the `|| {}` here? So `(headers || {}).to_h.merge(custom_headers)`. 

Perhaps better to let the `encryptor.encrypt` always return a hash. Not sure if that isn't already the case.

Reply:
I use headers.to_h specially to prevent the case and avoid `(hash || {})`

```ruby
{}.to_h => {}
nil.to_h => {}
```

Reply:
Ah yes, I didn't catch that!